### PR TITLE
Bug 1906722: Allow migrate after rollback

### PR DESCRIPTION
**Pull request by djwhatle**
Related to https://github.com/konveyor/mig-controller/issues/837 , must be tested with controller built off that PR.

https://bugzilla.redhat.com/show_bug.cgi?id=1906722

### CODE REVIEW

File: src/app/plan/duck/selectors.ts [hunk 1]
```diff
@@ -116,11 +116,28 @@ const getPlansWithPlanStatus = createSelector(
...
-      const latestType = latestMigration?.spec?.stage ? 'Stage' : 'Migration';
 
-      const hasSucceededStage = !!plan.Migrations.filter((m) => {
+      // Rollback acts as a "reset", so we should only consider migrations after rollback
+      let planMigrations;
+      if (plan.Migrations.length) {
+        const latestRollbackIndex = plan.Migrations.findIndex((m) => {
+          return m.spec.rollback && m.status.conditions.some((c) => c.type === 'Succeeded');
+        });
+        planMigrations = plan.Migrations.slice(0, latestRollbackIndex + 1);
```
Comment (djwhatle) [inline src/app/plan/duck/selectors.ts:126]: The major change this PR makes is that after rollback occurs, the list of `MigMigrations` considered when evaluating `hasMigrationSucceeded` etc. is limited to those that have taken place _after_ rollback.

**Comment by eriknelson** [comment]
Was looking at this BZ as I saw this email come through, @Derek assigned
you to this one I think they're linked:
https://bugzilla.redhat.com/show_bug.cgi?id=1906722

**Comment by djwhatle** [comment]
> Was looking at this BZ as I saw this email come through, @derek assigned you to this one I think they're linked: https://bugzilla.redhat.com/show_bug.cgi?id=1906722

Thanks for the link, I added details of the fix PRs into that BZ.

**Review by ibolton336 — APPROVED** [review]
LGTM!

**Pull request merged**